libs/platform/ebpf_object.h

```diff
@@ -25,10 +25,20 @@ extern "C"
     typedef void (*ebpf_free_object_t)(ebpf_core_object_t* object);
     typedef const ebpf_program_type_t (*ebpf_object_get_program_type_t)(_In_ const ebpf_core_object_t* object);
 
+    /**
+     * @brief Base object for all reference counted eBPF objects. This struct is embedded as the first entry in all
+     * reference counted eBPF objects. The reference count is 64bit to allow for atomic operations on 32bit and 64bit
+     * systems. The reference count is 8-byte aligned to avoid false sharing. The marker is used to detect
+     * use-after-free bugs. The marker is set to 'eobj' when the object is valid and is inverted when the object is
+     * freed. This allows for a fast check to see if the object is valid. The acquire_reference and release_reference
+     * functions are used to acquire and release a reference on the object.
+     */
     typedef struct _ebpf_base_object
     {
-        uint32_t marker;
-        volatile int32_t reference_count;
+        uint32_t marker; // Contains the 32bit value 'eobj' when the object is valid and is inverted when the object is
+                         // freed.
```

Comment:
The `marker` can have other values also:
https://github.com/microsoft/ebpf-for-windows/blob/750279068ccd78f6d4ebae530ed47c374df24f50/libs/execution_context/ebpf_native.c#L14